pg_shard--1.1--1.2.sql

```diff
@@ -1,13 +1,242 @@
--- add default values to id columns
-ALTER TABLE pgs_distribution_metadata.shard
-	ALTER COLUMN id
-	SET DEFAULT nextval('pgs_distribution_metadata.shard_id_sequence');
-ALTER TABLE pgs_distribution_metadata.shard_placement
-	ALTER COLUMN id
-	SET DEFAULT nextval('pgs_distribution_metadata.shard_placement_id_sequence');
-
--- associate sequences with their columns
-ALTER SEQUENCE pgs_distribution_metadata.shard_id_sequence
-	OWNED BY pgs_distribution_metadata.shard.id;
-ALTER SEQUENCE pgs_distribution_metadata.shard_placement_id_sequence
-	OWNED BY pgs_distribution_metadata.shard_placement.id;
+-- needed in our views
+CREATE FUNCTION column_to_column_name(table_oid oid, column_var text)
+RETURNS text
+AS 'MODULE_PATHNAME'
+LANGUAGE C STABLE STRICT;
+
+CREATE FUNCTION column_name_to_column(table_oid oid, column_name text)
+RETURNS text
+AS 'MODULE_PATHNAME'
+LANGUAGE C STABLE STRICT;
+
+DO $$
+DECLARE
+	use_citus_metadata boolean := false;
+	relation_name oid;
+BEGIN
+	BEGIN
+		PERFORM 'pg_catalog.pg_dist_partition'::regclass;
+		use_citus_metadata = true;
+	EXCEPTION
+		WHEN undefined_table THEN
+			use_citus_metadata = false;
+	END;
+
+	IF use_citus_metadata THEN
+		-- just in case, lock everyone out of pg_shard partitions
+		LOCK TABLE pgs_distribution_metadata.partition IN EXCLUSIVE MODE;
+		FOR relation_name IN SELECT relation_id::regclass::text
+		FROM pgs_distribution_metadata.partition LOOP
+			PERFORM sync_table_metadata_to_citus(relation_id::regclass);
+		END LOOP;
+
+		DROP TABLE pgs_distribution_metadata.partition,
+				   pgs_distribution_metadata.shard,
+				   pgs_distribution_metadata.shard_placement;
+
+		CREATE FUNCTION adapt_and_insert_shard() RETURNS TRIGGER AS $aais$
+		BEGIN
+			IF NEW.id IS NULL THEN
```

Comment:
Should we throw an exception if new.id isn't null? Is there a use-case we expect it wouldnt be NULL?